### PR TITLE
Added/updated main and rules pages

### PR DESCRIPTION
**Pull request by amsiglan**
Signed-off-by: Amardeepsingh Siglani <amardeep7194@gmail.com>

### Description
This PR
- Updates Main component to add routes to new pages in the plugin
- Adds UX for performing CRUD operations on Rules

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Rules/containers/Rules/components/Edit/Edit.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+              Cancel
+            </EuiButton>
+          </div>
+          <div style={{ marginRight: '10px' }}>
+            <EuiButton
+              type="submit"
+              fill
+              form="editForm"
+              // onClick={() => {history.push(ROUTES.RULES) }}
+              // disabled={!Boolean(Object.keys(Formikprops.errors).length === 0)}
```
Comment (AWSHurneyt) [inline public/pages/Rules/containers/Rules/components/Edit/Edit.tsx:69]: Not blocking for this draft, but can these comments be removed?

File: public/pages/Rules/containers/Rules/components/View/View.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        <EuiFormLabel>Rule Status</EuiFormLabel>
+        <div>{content.status}</div>
+
+        <EuiSpacer />
+
+        <EuiFormRow
+          label="Detection"
+          fullWidth
+          // helpText={Formikprops.touched.ruleDetection && Formikprops.errors.ruleDetection}
```
Comment (AWSHurneyt) [inline public/pages/Rules/containers/Rules/components/View/View.tsx:167]: Not blocking for this draft, but can this comment be removed?

**Review by AWSHurneyt — APPROVED** [review]

File: public/pages/Rules/containers/Rules/components/Edit/Edit.tsx (hunk 1 above)
Comment (amsiglan) [reply public/pages/Rules/containers/Rules/components/Edit/Edit.tsx:69]: Refactoring rules pages and will be removed

File: public/pages/Rules/containers/Rules/components/View/View.tsx (hunk 2 above)
Comment (amsiglan) [reply public/pages/Rules/containers/Rules/components/View/View.tsx:167]: refactoring rules code and will remove this in followup PR

File: public/pages/Rules/containers/Rules/components/Edit/Visual/Visual.tsx [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+                  value={Formikprops.values.ruleStatus}
+                />
+              </EuiFormRow>
+
+              <EuiSpacer />
+
+              <EuiFormRow
+                label="Detection"
+                fullWidth
+                // helpText={Formikprops.touched.ruleDetection && Formikprops.errors.ruleDetection}
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply public/pages/Rules/containers/Rules/components/Edit/Visual/Visual.tsx:324]: refactoring rules code and will remove this in followup PR

**Pull request merged**